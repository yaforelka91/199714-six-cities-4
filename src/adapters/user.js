const adaptUser = (user) => {
  return {
    id: user.id,
    email: user.email,
    name: user.name,
    picture: user.avatar_url,
    isSuper: user.is_pro,
  };
};

export default adaptUser;
