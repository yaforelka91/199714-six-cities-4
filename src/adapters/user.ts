import User from '../interfaces/user';

const adaptUser = (user): User => ({
  id: user.id,
  email: user.email,
  name: user.name,
  picture: user.avatar_url,
  isSuper: user.is_pro,
});

export default adaptUser;
