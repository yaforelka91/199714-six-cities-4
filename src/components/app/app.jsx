import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer.js';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import Main from '../main/main.jsx';
import OfferPage from '../offer-page/offer-page.jsx';
import {appTypes} from '../../types/types.js';
import {sortOffers} from '../../utils.js';

const MAX_CITIES_COUNT = 6;

class App extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      selectedOffer: null,
    };

    this._handleOfferTitleClick = this._handleOfferTitleClick.bind(this);
  }

  _handleOfferTitleClick(offer) {
    this.setState({
      selectedOffer: offer,
    });
  }

  _renderApp() {
    const {
      offersList,
      citiesList,
      city,
      onCityNameClick
    } = this.props;

    const {selectedOffer} = this.state;

    if (!selectedOffer) {
      return (
        <Main
          city={city}
          offersList={offersList}
          citiesList={citiesList}
          onOfferTitleClick={this._handleOfferTitleClick}
          onCityNameClick={onCityNameClick}
        />
      );
    }

    if (selectedOffer) {
      return (
        <OfferPage
          offer={selectedOffer}
          offersList={offersList}
          city={city}
          onOfferTitleClick={this._handleOfferTitleClick}
        />
      );
    }

    return null;
  }

  render() {
    const {offersList, city} = this.props;
    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/">
            {this._renderApp()}
          </Route>
          <Route exact path="/offer-page">
            <OfferPage
              offer={offersList[0]}
              offersList={offersList}
              city={city}
              onOfferTitleClick={this._handleOfferTitleClick}
            />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = appTypes;

const mapStateToProps = (state) => ({
  city: state.city,
  offersList: sortOffers(
      state.activeSorting,
      state.offersList
        .find((offer) => offer.city.id === state.city.id)
        .offers
  ),
  citiesList: state.offersList
    .map((offer) => offer.city)
    .slice(0, MAX_CITIES_COUNT),
});

const mapDispatchToProps = (dispatch) => ({
  onCityNameClick(city) {
    dispatch(ActionCreator.changeCity(city));
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
