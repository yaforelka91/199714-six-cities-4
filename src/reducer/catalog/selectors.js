import NameSpace from '../name-space.js';
import {getAllOffers, getCity} from '../data/selectors.js';
import {sortOffers} from '../../utils.js';
import {createSelector} from 'reselect';

const NAME_SPACE = NameSpace.CATALOG;

export const getSorting = (state) => {
  return state[NAME_SPACE].activeSorting;
};

export const getActiveCard = (state) => {
  return state[NAME_SPACE].activeCard;
};

export const getFilteredOffers = createSelector(
    getCity,
    getAllOffers,
    (activeCity, offers) => {
      return offers.filter((offer) => {
        return offer.city.name === activeCity.name;
      });
    }
);

export const getSortedOffers = createSelector(
    getSorting,
    getFilteredOffers,
    (sortType, filteredOffers) => {
      return sortOffers(sortType, filteredOffers);
    }
);
