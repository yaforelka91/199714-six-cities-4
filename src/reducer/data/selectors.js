import NameSpace from '../name-space.js';
import {createSelector} from 'reselect';

const NAME_SPACE = NameSpace.DATA;
const MAX_NEARBY_COUNT = 3;

export const getOffers = (state) => {
  return state[NAME_SPACE].offersList;
};

export const getCities = createSelector(
    getOffers,
    (offers) => {
      return [...new Set(offers
      .map((offer) => offer.city.name)
      .map((cityObject) => JSON.stringify(cityObject))
      )]
      .map((cityString) => JSON.parse(cityString));
    }
);

export const getNearbyOffers = (state) => {
  return state[NAME_SPACE].nearOffers.slice(0, MAX_NEARBY_COUNT);
};

export const getNearestOffers = createSelector(
    getOffers,
    getNearbyOffers,
    (allOffers, nearbyOffers) => {
      const reducedNearbyOffers = nearbyOffers.map((offer) => offer.id);

      return allOffers.filter((offer) => {
        return reducedNearbyOffers.includes(offer.id);
      });
    }
);

export const getError = (state) => {
  return state[NAME_SPACE].errorType;
};
