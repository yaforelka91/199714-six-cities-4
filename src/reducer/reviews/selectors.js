import NameSpace from '../name-space.js';
import {createSelector} from 'reselect';

const NAME_SPACE = NameSpace.REVIEWS;

export const getReviewsCount = (state) => state[NAME_SPACE].reviews.length;

export const getReviews = (state) => state[NAME_SPACE].reviews;

export const getSortedReviews = createSelector(
    getReviews,
    (reviews) =>
      reviews.sort(({visitTime: date1}, {visitTime: date2}) =>
        (new Date(date2)).valueOf() - (new Date(date1)).valueOf()
      )
);
