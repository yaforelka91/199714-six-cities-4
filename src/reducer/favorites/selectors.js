import NameSpace from '../name-space.js';
import {createSelector} from 'reselect';
import {getOffers} from '../data/selectors.js';

const NAME_SPACE = NameSpace.FAVORITES;

export const getFavorites = (state) => {
  return state[NAME_SPACE].favoriteList;
};

export const getFavoriteOffers = createSelector(
    getOffers,
    getFavorites,
    (allOffers, favoriteOffers) => {
      const reducedFavoriteOffers = favoriteOffers.map((offer) => offer.isFavorite);

      return allOffers.filter((offer) => {
        return reducedFavoriteOffers.includes(offer.isFavorite);
      });
    }
);

export const getGroupedCities = createSelector(
    getFavoriteOffers,
    (offers) => {
      return [...new Set(offers
  .map((offer) => offer.city.name)
  .map((cityObject) => JSON.stringify(cityObject))
      )]
  .map((cityString) => JSON.parse(cityString));
    }
);

export const getGroupedFavoriteOffers = createSelector(
    getGroupedCities,
    getFavoriteOffers,
    (cities, offers) => {
      return cities.map((city) => {
        return {
          [city]: offers.filter((offer) => {
            return offer.city.name === city;
          }),
        };
      });
    }
);
