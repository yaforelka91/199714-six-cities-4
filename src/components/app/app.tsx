import React from 'react';
import {connect} from 'react-redux';
import {Router, Switch, Route, Redirect} from 'react-router-dom';
import {History} from 'history';
import Page from '../page/page';
import Main from '../main/main';
import OfferPage from '../offer-page/offer-page';
import Login from '../login/login';
import {getAuthorizationStatus, getAuthorizationProgress} from '../../reducer/user/selectors';
import {AuthorizationStatus} from '../../reducer/user/user';
import {AppRoute} from '../../types';
import {getLoadingStatus, getError, getOffers} from '../../reducer/data/selectors';
import Favorites from '../favorites/favorites';
import PrivateRoute from '../private-route/private-route';
import {getGroupedFavoriteOffers} from '../../reducer/favorites/selectors';
import ErrorScreen from '../error-screen/error-screen';
import NoPlaces from '../no-places/no-places';
import Offer from '../../interfaces/offer';

type Props = {
  authorizationStatus: string;
  favoriteOffers: {
    [key: string]: Offer[];
  }[];
  history: History;
  offers: Offer[];
  isAuthorizationInProgress?: boolean;
  isOffersLoading?: boolean;
  errorType?: string;
}

const App: React.FC<Props> = (props: Props) => {
  const {
    authorizationStatus,
    favoriteOffers,
    history,
    offers,
    isAuthorizationInProgress = false,
    isOffersLoading = false,
    errorType = ``,
  } = props;

  if (!isOffersLoading && !isAuthorizationInProgress && errorType === `` && offers.length === 0) {
    return (
      <div className="page page--gray page--main">
        <main className="page__main page__main--index page__main--index-empty">
          <div className="cities">
            <div className="cities__places-container container">
              <NoPlaces />
              <div className="cities__right-section" />
            </div>
          </div>
        </main>
      </div>
    );
  }

  return (
    <Router history={history}>
      <Switch>
        <Route exact path={AppRoute.ROOT} render={() => {
          return (
            <Page
              className='page--gray page--main'
              errorMessage={errorType}
              isLoading={isAuthorizationInProgress || isOffersLoading}
              renderPage={() => <Main />}
            />
          );
        }}
        />
        <Route exact path={`${AppRoute.OFFER}/:id`} render={({match}) => {
          return (
            <Page
              errorMessage={errorType}
              isLoading={isAuthorizationInProgress || isOffersLoading}
              renderPage={() => {
                const offer = offers.find((hotel) => hotel.id === +match.params.id);

                if (offer) {
                  return (
                    <OfferPage
                      authorizationStatus={authorizationStatus}
                      offer={offer}
                    />
                  );
                } else {
                  return <ErrorScreen message='offer does not exist' />;
                }
              }}
            />
          );
        }}
        />
        <Route exact path={AppRoute.LOGIN} render={() => {
          if (authorizationStatus === AuthorizationStatus.AUTH) {
            return <Redirect to={AppRoute.ROOT} />;
          }

          return (
            <Page
              className='page--gray page--login'
              errorMessage={errorType}
              isLoading={isAuthorizationInProgress || isOffersLoading}
              renderPage={() => <Login />}
            />
          );
        }}
        />
        <PrivateRoute
          exact
          path={AppRoute.FAVORITES}
          authorizationStatus={authorizationStatus}
          isAuthorizationInProgress={isAuthorizationInProgress}
          render={() => {
            return (
              <Page
                className={`${favoriteOffers.length === 0 ? `page--favorites-empty` : ``}`}
                hasFooter={true}
                errorMessage={errorType}
                isLoading={isAuthorizationInProgress || isOffersLoading}
                renderPage={() => <Favorites />}
              />
            );
          }}
        />
        <Route
          render={() => (
            <Page
              errorMessage={errorType}
              isLoading={isAuthorizationInProgress || isOffersLoading}
              renderPage={() => <ErrorScreen message='page is not exist' />}
            />
          )}
        />
      </Switch>
    </Router>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  favoriteOffers: getGroupedFavoriteOffers(state),
  offers: getOffers(state),
  isAuthorizationInProgress: getAuthorizationProgress(state),
  isOffersLoading: getLoadingStatus(state),
  errorType: getError(state),
});

export {App};
export default connect(mapStateToProps)(App);
