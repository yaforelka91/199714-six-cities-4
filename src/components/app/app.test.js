import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {App} from './app.jsx';
import NameSpace from '../../reducer/name-space.js';
import {getActiveOffer, getNearestOffers, getFilteredOffers} from '../../reducer/catalog/selectors.js';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {getAuthorizationStatus} from '../../reducer/user/selectors.js';

const mockStore = configureStore([]);

const offersList = [
  {
    city: {
      name: `Dusseldorf`,
      coords: [51.230569, 6.787428],
      zoom: 1,
    },
    id: 8,
    coords: [51.202025, 6.800942],
    title: `Apartment in Dusseldorf`,
    description: [
      `A quiet cozy and picturesque that hides behind a 
          river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
      `An independent House, strategically located 
          between Rembrand Square and National Opera, 
          but where the bustle of the city comes to rest 
          in this alley flowery and colorful.`
    ],
    picture: `http://placeimg.com/260/200/arch`,
    pictures: [
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`
    ],
    price: 180,
    type: `Apartment`,
    isPremium: true,
    isFavorite: false,
    rating: 4.9,
    bedrooms: 4,
    guests: 6,
    services: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      id: 2,
      name: `Bob`,
      picture: `http://placekitten.com/74/74`,
      isSuper: true,
    },
  },
  {
    city: {
      name: `Hamburg`,
      coords: [53.552645, 9.966287],
      zoom: 1,
    },
    id: 7,
    coords: [53.553542, 9.912283],
    title: `Apartment in Hamburg`,
    description: [
      `A quiet cozy and picturesque that hides behind a 
          river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
      `An independent House, strategically located 
          between Rembrand Square and National Opera, 
          but where the bustle of the city comes to rest 
          in this alley flowery and colorful.`
    ],
    picture: `http://placeimg.com/260/200/arch`,
    pictures: [
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`
    ],
    price: 180,
    type: `Apartment`,
    isPremium: true,
    isFavorite: false,
    rating: 4.9,
    bedrooms: 4,
    guests: 6,
    services: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      id: 1,
      name: `Bob`,
      picture: `http://placekitten.com/74/74`,
      isSuper: true,
    },
  },
  {
    city: {
      name: `Dusseldorf`,
      coords: [51.230569, 6.787428],
      zoom: 1,
    },
    id: 9,
    coords: [51.202045625, 6.800456942],
    title: `Apartment in Dusseldorf 2`,
    description: [
      `A quiet cozy and picturesque that hides behind a 
          river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
      `An independent House, strategically located 
          between Rembrand Square and National Opera, 
          but where the bustle of the city comes to rest 
          in this alley flowery and colorful.`
    ],
    picture: `http://placeimg.com/260/200/arch`,
    pictures: [
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`,
      `http://placeimg.com/260/200/arch`
    ],
    price: 190,
    type: `Apartment`,
    isPremium: true,
    isFavorite: false,
    rating: 4.9,
    bedrooms: 4,
    guests: 6,
    services: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    host: {
      id: 3,
      name: `Mike`,
      picture: `http://placekitten.com/74/74`,
      isSuper: true,
    },
  },
];

const userData = {
  email: `test@test.com`,
  picture: `/pic1.jpg`
};

describe(`AppSnapshot`, () => {
  it(`should render Main page`, () => {
    const store = mockStore({
      [NameSpace.CATALOG]: {
        activeCard: -1,
        activeCity: offersList[0].city.name,
      },
      [NameSpace.DATA]: {
        offersList,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userData,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <App
            authorizationStatus={getAuthorizationStatus(store.getState())}
            login={() => {}}
            userData={userData}
            offersList={getFilteredOffers(store.getState())}
            activeCard={getActiveOffer(store.getState())}
            onOfferTitleClick={() => {}}
          />
        </Provider>,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render Offer page`, () => {
    const store = mockStore({
      [NameSpace.CATALOG]: {
        activeCard: 8,
        activeCity: `Dusseldorf`,
      },
      [NameSpace.DATA]: {
        offersList,
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
        userData,
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <App
            authorizationStatus={getAuthorizationStatus(store.getState())}
            login={() => {}}
            userData={userData}
            activeCard={getActiveOffer(store.getState())}
            offersList={getNearestOffers(store.getState())}
            onOfferTitleClick={() => {}}
          />
        </Provider>,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
