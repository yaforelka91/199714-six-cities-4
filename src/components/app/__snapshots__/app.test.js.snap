// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSnapshot should render Main page 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Hamburg
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
             
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom "
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option "
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
            <select
              className="places__sorting-type"
              id="places-sorting"
              onChange={[Function]}
              style={
                Object {
                  "border": 0,
                  "clip": "rect(1px, 1px, 1px, 1px)",
                  "height": "1px",
                  "overflow": "hidden",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              value="popular"
            >
              <option
                className="places__option"
                value="popular"
              >
                Popular
              </option>
              <option
                className="places__option"
                value="to-high"
              >
                Price: low to high
              </option>
              <option
                className="places__option"
                value="to-low"
              >
                Price: high to low
              </option>
              <option
                className="places__option"
                value="top-rated"
              >
                Top rated first
              </option>
            </select>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="http://placeimg.com/260/200/arch"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      180
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Apartment in Hamburg
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`AppSnapshot should render Offer page 1`] = `
<div
  className="page "
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="http://placeimg.com/260/200/arch"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Apartment in Hamburg
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.9
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              4 Bedroom
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 6 adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              180
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Coffee machine
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
              <li
                className="property__inside-item"
              >
                Dishwasher
              </li>
              <li
                className="property__inside-item"
              >
                Cabel TV
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="http://placekitten.com/74/74"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Bob
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a 
          river by the unique lightness of Amsterdam.
          The building is green and from 18th century.
              </p>
              <p
                className="property__text"
              >
                An independent House, strategically located 
          between Rembrand Square and National Opera, 
          but where the bustle of the city comes to rest 
          in this alley flowery and colorful.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="http://placekitten.com/54/54"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Alice
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2016-12-30"
                  >
                    December 2016
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="places__list near-places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
